### ISSUE TITLE
Clippy document test meet compile error

### ISSUE THREAD

**Issue by lwz23** [rust-lang/rust-clippy#13520]
### Summary

When I tried to experiment with the Clippy Getting Started documentation（https://doc.rust-lang.org/nightly/clippy/development/adding_lints.html）, I couldn't pass compile.

### Reproducer

I tried this code:

```rust
use rustc_ast::ast::*;
use rustc_lint::{EarlyContext, EarlyLintPass};
use rustc_session::declare_lint_pass;

declare_clippy_lint! {
    /// ### What it does
    ///
    /// ### Why is this bad?
    ///
    /// ### Example
    /// ```no_run
    /// // example code where clippy issues a warning
    /// ```
    /// Use instead:
    /// ```no_run
    /// // example code which does not raise clippy warning
    /// ```
    #[clippy::version = "1.83.0"]
    pub FOO_FUNCTIONS,
    pedantic,
    "default lint description"
}

declare_lint_pass!(FooFunctions => [FOO_FUNCTIONS]);

impl EarlyLintPass for FooFunctions {
    fn check_fn(&mut self, cx: &EarlyContext<'_>, fn_kind: FnKind<'_>, span: Span, _: NodeId) {
        // TODO: Emit lint here
    }
}

```

I expected to see this happen:
```
pass compile
```
Instead, this happened:
```
error[E0412]: cannot find type `FnKind` in this scope
  --> clippy_lints\src\foo_functions.rs:27:60
   |
27 |     fn check_fn(&mut self, cx: &EarlyContext<'_>, fn_kind: FnKind<'_>, span: Span, _: NodeId) {
   |                                                            ^^^^^^
  --> C:\Users\ROG\.rustup\toolchains\nightly-2024-10-03-x86_64-pc-windows-msvc\lib/rustlib/src/rust\compiler\rustc_ast\src\ast.rs:2155:1
   |
   = note: similarly named enum `TyKind` defined here
   |
help: an enum with a similar name exists
   |
27 |     fn check_fn(&mut self, cx: &EarlyContext<'_>, fn_kind: TyKind<'_>, span: Span, _: NodeId) {
   |                                                            ~~~~~~
help: consider importing one of these enums
   |
1  + use rustc_ast::mut_visit::FnKind;
   |
1  + use rustc_ast::visit::FnKind;
   |
1  + use rustc_hir::intravisit::FnKind;
   |

error[E0412]: cannot find type `Span` in this scope
  --> clippy_lints\src\foo_functions.rs:27:78
   |
27 |     fn check_fn(&mut self, cx: &EarlyContext<'_>, fn_kind: FnKind<'_>, span: Span, _: NodeId) {
   |                                                                              ^^^^ not found in this scope
   |
help: consider importing one of these structs
   |
1  + use regex_syntax::ast::Span;
   |
1  + use rustc_span::Span;
   |

For more information about this error, try `rustc --explain E0412`.
error: could not compile `clippy_lints` (lib) due to 2 previous errors
```

**Comment by lwz23** [comment]
I'm not sure if the documentation wasn't up to date, but this is the basic beginner documentation for Clippy, and it would have compiled if I hadn't copied the code by mistake.

**Comment by blyxyas** [comment]
Closing as I assume you managed to fix it :)
